### ISSUE TITLE
Detect stuck inf in group actions

### ISSUE THREAD

**Issue by billw2012** [Sparker95/Vindicta#449]
Add StuckDetector general class for monitoring position over time to detect when it isn't changing fast enough. Use it for move actions etc.

**Comment by Sparker95** [comment]
Good idea, I think it makes sense to make the stuck detector a separate component. Now 'stuck detection' is embedded in different 'move' actions.
